homeassistant/components/deconz/climate.py

```diff
@@ -231,16 +244,23 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
     @property
     def current_temperature(self) -> float:
         """Return the current temperature."""
+        assert isinstance(self._device.temperature, float)
         return self._device.temperature
 
     @property
-    def target_temperature(self) -> float:
+    def target_temperature(self) -> float | None:
         """Return the target temperature."""
-        if self._device.mode == THERMOSTAT_MODE_COOL:
+        if self._device.mode == THERMOSTAT_MODE_COOL and self._device.cooling_setpoint:
+            assert isinstance(self._device.cooling_setpoint, float)
             return self._device.cooling_setpoint
-        return self._device.heating_setpoint
 
-    async def async_set_temperature(self, **kwargs):
+        if self._device.heating_setpoint:
+            assert isinstance(self._device.heating_setpoint, float)
+            return self._device.heating_setpoint
+
+        return None
```

Comment:
Change of behaviour? (See comment above)

Reply:
Same response as above

Reply:
Removed assert